src/sinks/datadog/metrics/integration_tests.rs

```diff
@@ -104,16 +126,105 @@ async fn smoke() {
 
     assert!(output.len() == 1, "Should have received a response");
 
-    let val = output.first().unwrap();
+    let request = output.first().unwrap();
+
+    match request.0.uri.path() {
+        SERIES_V1_PATH => validate_json(request),
+        SERIES_V2_PATH => validate_protobuf(request),
+        _ => panic!("Unexpected request type received!"),
+    }
+}
 
+fn validate_common(request: &(Parts, Bytes)) {
+    assert_eq!(request.0.headers.get("DD-API-KEY").unwrap(), "atoken");
+    assert!(request.0.headers.contains_key("DD-Agent-Payload"));
+}
+
+fn validate_protobuf(request: &(Parts, Bytes)) {
     assert_eq!(
-        val.0.headers.get("Content-Type").unwrap(),
+        request.0.headers.get("Content-Type").unwrap(),
+        "application/x-protobuf"
+    );
+
+    validate_common(request);
+
+    let compressed_payload = request.1.to_vec();
+    let payload = decompress_payload(compressed_payload).expect("Could not decompress payload");
+    let frame = Bytes::copy_from_slice(&payload);
+
+    let payload =
+        ddmetric_proto::MetricPayload::decode(frame).expect("Could not decode protobuf frame");
+
+    let series = payload.series;
+
+    assert!(!series.is_empty());
+
+    // check metrics are sorted by name, which helps HTTP compression
+    let metric_names: Vec<String> = series.iter().map(|serie| serie.metric.clone()).collect();
+    let mut sorted_names = metric_names.clone();
+    sorted_names.sort();
+    assert_eq!(metric_names, sorted_names);
```

Comment:
💭 Is this true for proto payloads as well? Currently, we have two reasons for why we sort the metrics in a payload: (1) for improved compression and (2) to aggregate counters. In the future, when we revamp the aggregation logic, the second reason will be obsolete. And if this doesn't hold true for proto payloads either, then we can remove the need to sort payloads altogether when we remove support for the V1 endpoint.

Reply:
Mmm yeah, naively I would think that the sort by name wouldn't matter for the proto case. I kept the "check" just to be consistent but I think it makes sense to remove it from the v2 checks, and then if/when we remove the v1 support, we'll remove the v1 test case here. WDYT?

Reply:
My hunch is that the performance gains still apply for the proto case because it also use zlib compression.